File: DevExpress-Demo/CS/DiagramDemo.Wpf/DemoTesting/Fixtures.cs
using DevExpress.Xpf.DemoBase.DemoTesting;
using System.Threading;
using System.Windows.Threading;
using System;
using DevExpress.Xpf.Editors;
using DevExpress.Xpf.Editors.Helpers;
using System.Windows;
using System.Globalization;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.DemoBase;
using System.Linq;

namespace DiagramDemo.Tests {
    public class DiagramCheckAllDemosFixture : CheckAllDemosFixture {
        protected override bool AllowSwitchToTheTheme(Type moduleType, Theme theme) {
            return theme != Theme.HybridApp;
        }
    }
    public class DiagramDemoModulesAccessor : DemoModulesAccessor<DiagramDemoModule> {
        public DiagramDemoModulesAccessor(BaseDemoTestingFixture fixture)
            : base(fixture) {
        }
    }
    public abstract class BaseDiagramDemoTestingFixture : BaseDemoTestingFixture {
        protected DiagramDemoModulesAccessor ModuleAccessor { get; private set; }
        public BaseDiagramDemoTestingFixture() {
            ModuleAccessor = GetModulesAccessor();
        }
        protected abstract DiagramDemoModulesAccessor GetModulesAccessor();
    }
}

File: DevExpress-Demo/VB/DiagramDemo.Wpf/DemoTesting/Fixtures.vb
Imports DevExpress.Xpf.DemoBase.DemoTesting
Imports System
Imports DevExpress.Xpf.Editors.Helpers
Imports DevExpress.Xpf.Core

Namespace DiagramDemo.Tests

    Public Class DiagramCheckAllDemosFixture
        Inherits CheckAllDemosFixture

        Protected Overrides Function AllowSwitchToTheTheme(ByVal moduleType As Type, ByVal theme As Theme) As Boolean
            Return theme IsNot Theme.HybridApp
        End Function
    End Class

    Public Class DiagramDemoModulesAccessor
        Inherits DemoModulesAccessor(Of DiagramDemoModule)

        Public Sub New(ByVal fixture As BaseDemoTestingFixture)
            MyBase.New(fixture)
        End Sub
    End Class

    Public MustInherit Class BaseDiagramDemoTestingFixture
        Inherits BaseDemoTestingFixture

        Private _ModuleAccessor As DiagramDemoModulesAccessor

        Protected Property ModuleAccessor As DiagramDemoModulesAccessor
            Get
                Return _ModuleAccessor
            End Get

            Private Set(ByVal value As DiagramDemoModulesAccessor)
                _ModuleAccessor = value
            End Set
        End Property

        Public Sub New()
            ModuleAccessor = GetModulesAccessor()
        End Sub

        Protected MustOverride Function GetModulesAccessor() As DiagramDemoModulesAccessor
    End Class
End Namespace
